src/screens/ProfileScreen/elements/KarmaLock.tsx

```diff
@@ -0,0 +1,126 @@
+        </Text>
+        <Tooltip
+          allowChildInteraction={true}
+          isVisible={isTooltipShown}
+          placement={'bottom'}
+          backgroundColor={COLORS.black}
+          closeOnContentInteraction={false}
+          onClose={() => setIsTooltipShown(false)}
+          closeOnBackgroundInteraction={true}
+          contentStyle={{
+            borderRadius: 10,
+            padding: 16,
+            height: 'auto'
+          }}
+          content={
+            <View>
+              <Text
+                style={{
+                  fontWeight: '600',
+                  fontSize: normalize(16),
+                  lineHeight: normalize(24),
+                  marginBottom: normalize(8)
+                }}>
+                What is my Karma Score?
+              </Text>
+              <Text
+                style={{
+                  fontWeight: '400',
+                  fontSize: normalize(12),
+                  lineHeight: normalize(18),
+                  marginBottom: normalize(8),
+                  color: '#69707C'
+                }}>
+                The higher your Karma score, the higher your visibility on the platform. Get
+                rewarded for positive contributions to the community - and avoid being blocked by
+                others! Your Karma score will be shown as a circle around your profile picture,
+                including for your anonymous posts.
+              </Text>
+              <Text
+                style={{
+                  fontWeight: '600',
+                  paddingVertical: normalize(8),
+                  fontSize: normalize(14),
+                  lineHeight: normalize(20),
+                  color: '#4782D7',
+                  width: '100%',
+                  textAlign: 'right'
+                }}
+                onPress={() => setIsTooltipShown(false)}>
+                Got it!
+              </Text>
+            </View>
+          }>
+          <TouchableOpacity onPress={() => setIsTooltipShown(true)}>
+            <View>
+              <MemoIcQuestionMark width={normalize(13)} height={normalize(13)} />
+            </View>
+          </TouchableOpacity>
+        </Tooltip>
+      </View>
+      <TouchableOpacity
+        onPress={props.onPressCreatePost}
+        style={{
+          backgroundColor: COLORS.signed_primary,
+          height: 34,
+          marginTop: 8,
+          borderRadius: 8
+        }}>
+        <View
+          style={{
+            display: 'flex',
+            flexDirection: 'row',
+            justifyContent: 'center',
+            alignItems: 'center',
+            marginVertical: 8,
+            marginHorizontal: 10
+          }}>
+          <PencilIcon />
+          <Text
+            style={{
+              fontSize: 12,
+              color: COLORS.white,
+              marginLeft: 8
+            }}>
+            Start Posting
+          </Text>
+        </View>
+      </TouchableOpacity>
+    </View>
+  );
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `KarmaLock` component is quite large and contains a lot of inline styles. This can make it difficult to read and maintain. Consider extracting the styles into a separate StyleSheet object or even a separate file. Also, consider breaking down this component into smaller, more manageable sub-components if possible. Here's an example of how you could refactor the styles:

```diff
- 14:       style={{
- 15:         backgroundColor: COLORS.gray100,
- 16:         borderRadius: 12,
- 17:         paddingVertical: 8,
- 18:         paddingHorizontal: 4
- 19:       }}>
+ 14:       style={styles.container}>

// ...

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.gray100,
    borderRadius: 12,
    paddingVertical: 8,
    paddingHorizontal: 4,
  },
  // ... other styles ...
});
```

Also, remember to import StyleSheet from 'react-native' at the top of your file.

```javascript
import { StyleSheet } from 'react-native';
```
This will not only improve readability but also potentially improve performance as the styles are processed once and cached, rather than being processed on every render.


<!-- This is an auto-generated comment by OSS CodeRabbit -->